src/dotnet-install.ps1

```diff
@@ -778,7 +835,19 @@ catch {
 }
 
 if ($DownloadFailed) {
-    throw "Could not find/download: `"$assetName`" with version = $SpecificVersion`nRefer to: https://aka.ms/dotnet-os-lifecycle for information on .NET Core support"
+    if (($PrimaryDownloadStatusCode -eq "404") -and ((-not $LegacyDownloadLink) -or ($LegacyDownloadStatusCode -eq "404"))) {
```

Comment:
nit: you sometimes compare the status code as string and sometimes as number - line 822), is that intentional?